### ISSUE TITLE
Complicated specs from variant config not working

### ISSUE THREAD

**Issue by wolfv** [prefix-dev/rattler-build#323]
I got an issue when usign this variant_config.yaml with a recipe that depends on `python`.

```yaml
python:
- 3.10.* *_cpython
- 3.11.* *_cpython
- 3.12.* *_cpython
```

The complaint is:

```
✔ linking              [00:00:01] Done!                                                                                                            thread 'main' panicked at 'Invalid version spec: InvalidVersionTree(ParseError("0: at line 1, in Eof:\n3.11.* *_cpython\n      ^\n\n"))', src/render/resolved_dependencies.rs:245:70
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by wolfv** [comment]
This is fixed, but still needs to expose the StringMatcherParserError in rattler.

**Comment by wolfv** [comment]
fixed in #353
